### ISSUE TITLE
produce valid BED records for short sequences

### ISSUE THREAD

**Issue by ptrebert** [PacificBiosciences/HiFiCNV#29]
Hi,
I am using hificnv v1.0 (also tested with the previous release) and have a couple of invalid BED records in the `.copynum.bedgraph` output, e.g.,

```
chrUn_KI270378v1        0       0       0
chrUn_KI270379v1        0       0       0
chrUn_KI270389v1        0       0       0
```

where all of the above sequences have a length below the default (unchangeable?) bin size: `The bin size is 2kb` as per your docs.
I suspect this is causing the issue.

I see (assume ...) that the option `--exclude <FILE>` is an obvious workaround for that problem, but I would suggest to handle the above case in a way that does not lead to records with a 0 for start and end in the output (or, at least, to add a hint to the quickstart FAQ in the docs).

Thanks for considering.

Best,
Peter

**Comment by holtjma** [comment]
Hi Peter,

> I am using hificnv v1.0 (also tested with the previous release) and have a couple of invalid BED records in the .copynum.bedgraph output

Can you elaborate on what is invalid in the records?  Our understanding based on the [BED file format](https://genome.ucsc.edu/FAQ/FAQformat.html#format1) is that records at the start of the chromosome _should_ start with 0 and that 0-length records are valid records.  While these records are not particularly informative, I don't think they should break any software that is following the BED specification.

As for those chromosomes in question, I do not recommend using the results from HiFiCNV. The software is tuned to work on the core human chromosomes and not on ALT or unplaced contigs that might be in your reference genome.

Matt

**Comment by ptrebert** [comment]
Hi Matt,
thanks for your comment. I had never realized that there is this remark about 0-length records in the BED format spec, and its intended (?) use for insertion calls also surprises me a bit; I usually see insertions encoded as (pos, pos+1) in BED formatted files, but that is just my sample of the world, of course.

The tool that breaks with that input is `bedtools [intersect]` (v2.31). Looking at one of its competitors (bedops), I would also guess that it enforces a minimum region length of 1 bp given the docs of its utility tool `vcf2bed`:

```
[...]
we perform the equivalent of a single-base insertion to make BED output that is guaranteed to work with BEDOPS
[...]
When using --deletions, the stop value of the BED output is determined
by the length difference between ALT and REF alleles.
Use of --insertions or --snvs yields a one-base BED element.
```
[ source: bedops.readthedocs.io/en/latest/content/reference/file-management/conversion/vcf2bed.html ]

So, in conclusion, yes, you are right, the records are not invalid according to the BED specification and arguing about common practice is of course always subjective. I guess my request would thus be to simply pass records with a length below the 2 kbp bin size unchanged into the output bedgraph (with some default copy number value). Feel free to close.

**Comment by holtjma** [comment]
No worries, thanks for raising the issue!  I'll make a note of it, and maybe there's something we can do to make these short records disappear from the outputs.

Some additional info for my own future reference, the error will look something like this:
```
$ bedtools intersect -a ./a.bedgraph -b ./b.bedgraph 
ERROR: Received illegal bin number -1 from getBin call.
Maximum values is: 2396745
This typically means that your coordinates are
negative or too large to represent in the data
structure bedtools uses to find intersections.ERROR: Unable to add record to tree.
```

This also appears to be a known bedtools issue that was introduced sometime after v2.25.0: https://github.com/arq5x/bedtools2/issues/869
